h2/src/main/org/h2/command/Parser.java

```diff
@@ -5956,6 +5957,17 @@ private DefineCommand parseAlterTableAddConstraintIf(String tableName,
                 read("BTREE");
             }
             return command;
+        } else if (allowAffinityKey && readIf("AFFINITY")) {
+            read("KEY");
+            IndexColumn[] indexColumns = new IndexColumn[] { new IndexColumn() };
+            indexColumns[0].columnName = readColumnIdentifier();
```

Comment:
So you suggest that we use `parseIndexColumnList()` here like parsing of PRIMARY KEY does? If yes, then this would imply that we allow various indexes specific stuff (asc/desc, nulls first/last) where it actually does not belong and has no actual meaning (and MemSQL allows just list of names here). If `parseIndexColumnList()` is not what you meant, then please clarify.